mods/cnc/maps/twist-of-fate/twist-of-fate.lua

```diff
@@ -0,0 +1,327 @@
+
+SendNodAirstrike = function(loop)
+	if AstkDelay == 0 then
+		return
+	end
+	local target = GetAirstrikeTarget(Player1)
+	if target then
+		NodAirSupport.TargetAirstrike(target, Angle.SouthEast)
+		if loop then
+			Trigger.AfterDelay(DateTime.Seconds(AstkDelay), function() SendNodAirstrike(true) end)
+		end
+	else
+		Trigger.AfterDelay(DateTime.Seconds(20), function() SendNodAirstrike(loop) end)
+	end
+end
+
+SendEasyReinforcements = function(i)
+	local team = { }
+	if i < 4 then
+		team = ReinforceTeams[i]
+	else
+		team = ReinforceTeams[Utils.RandomInteger(1, 4)]
+	end
+	Media.PlaySpeechNotification(Player1, "Reinforce")
+	Reinforcements.Reinforce(Player1, team, { CPos.New(56,2), waypoint0.Location }, 25, function(a)
+		a.Move(waypoint1.Location, 2)
+		a.Move(waypoint2.Location, 2)
+		a.Move(CPos.New(49,44), 2)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(ReinforceDelay), function()
+		SendEasyReinforcements(i + 1)
+	end)
+end
+
+CheckObjectives = function()
+	if EnemyAi.HasNoRequiredUnits() then Player1.MarkCompletedObjective(EliminateAllNod) end
+	if Player1.HasNoRequiredUnits() then Player1.MarkFailedObjective(EliminateAllNod) end
+	Trigger.AfterDelay(50, CheckObjectives)
+end
+
+CompleteCaptureCommCenterObjective = function()
+	Player1.MarkCompletedObjective(CaptureCommCenter)
+	if not NodCYard.IsDead and NodCYard.Owner == EnemyAi then
+		Trigger.Clear(NodCYard, "OnDamaged")
+		Trigger.AfterDelay(1, function()
+			RepairBuilding(EnemyAi, NodCYard, 0.75)
+			NodCYard.StartBuildingRepairs()
+		end)
+	end
+	Media.DisplayMessage(UserInterface.Translate("communications-center-captured-sams-located"))
+	local activeSams = EnemyAi.GetActorsByType("sam")
+	local miniCams = { }
+	if #activeSams > 0 then
+		Utils.Do(activeSams, function(s)
+			table.insert(miniCams, Actor.Create("camera.mini", true, { Owner = Player1, Location = s.Location }))
+			table.insert(miniCams, Actor.Create("camera.mini", true, { Owner = Player1, Location = CPos.New(s.Location.X + 1, s.Location.Y) }))
+		end)
+		Trigger.AfterDelay(1, function()
+			Utils.Do(miniCams, function(c)
+				c.Destroy()
+			end)
+		end)
+		Trigger.AfterDelay(200, function()
+			DestroySams = AddSecondaryObjective(Player1, "destroy-sams")
+			activeSams = EnemyAi.GetActorsByType("sam")
+			if #activeSams > 0 then
+				Trigger.OnAllKilled(activeSams, function()
+					AirSupport = Actor.Create("Astk.proxy", true, { Owner = Player1 })
+					Player1.MarkCompletedObjective(DestroySams)
+				end)
+			else
+				AirSupport = Actor.Create("Astk.proxy", true, { Owner = Player1 })
+				Player1.MarkCompletedObjective(DestroySams)
+			end
+		end)
+	else
+		Trigger.AfterDelay(200, function()
+			DestroySams = AddSecondaryObjective(Player1, "destroy-sams")
+			AirSupport = Actor.Create("Astk.proxy", true, { Owner = Player1 })
+			Player1.MarkCompletedObjective(DestroySams)
+		end)
+	end
+end
+
+WorldLoaded = function()
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	Player1 = Player.GetPlayer("GDI")
```

Comment:
This should be `GDI = Player.GetPlayer("GDI")` to match the other scripts. (Also see #20698.)